### PR TITLE
Documented the interaction between Backpack and Template Haskell.

### PR DESCRIPTION
**Pull request by danidiaz**
This adds documentation for #7353.

### CODE REVIEW

File: doc/cabal-package.rst [hunk 1]
```diff
@@ -3452,6 +3452,14 @@ create many separate Cabal packages.  You may also find it useful to use
 libraries to Backpack-unware users (e.g., Backpack can be used entirely
 as an implementation detail.)
 
+Backpack imposes a limitation on Template Haskell that goes beyond the usual TH
+stage restriction: it's not possible to splice TH code imported from a library
+that is still "indefinite", that is, a library for which some module signatures
+still haven't been matched with implementations. The reason is that indefinite
+libraries are typechecked, but not compiled, so there's no actual TH code to
+run while splicing. Splicing TH code from a non-indefinite package into an
```
Comment (phadej) [inline doc/cabal-package.rst:3460]: non-indefinite -> definite

Also avoid word library and package here, these are units or components.

Comment (danidiaz) [reply doc/cabal-package.rst]: Updated.

**Review by emilypi — APPROVED** [review]
Thanks @danidiaz this looks good.

**Comment by emilypi** [comment]
@danidiaz  CI is fixed if you rebase to master

**Pull request merged**